dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java

```diff
@@ -271,45 +255,42 @@ public String formatPeriod(Period period, boolean shortVersion) {
 
     PeriodType periodType = period.getPeriodType();
     String typeName = periodType.getName();
+    Calendar calendar = PeriodType.getCalendar();
+
+    if (periodType instanceof WeeklyAbstractPeriodType) {
+      DateTimeUnit start;
+      DateTimeUnit end;
+      String week = null;
+
+      if (calendar.isIso8601()) {
+        start = DateTimeUnit.fromJdkDate(period.getStartDate());
+        end = DateTimeUnit.fromJdkDate(period.getEndDate());
+        LocalDate startDate =
+            Instant.ofEpochMilli(period.getStartDate().getTime())
+                .atZone(ZoneId.systemDefault())
+                .toLocalDate();
+        WeekFields weekFields =
+            WeekFields.of(PeriodType.MAP_WEEK_TYPE.get(periodType.getName()), 4);
+        week = String.valueOf(startDate.get(weekFields.weekOfWeekBasedYear()));
 
-    if (periodType instanceof WeeklyAbstractPeriodType) // Use ISO dates
-    // due to
-    // potential week
-    // confusion
-    {
-      DateTime dateTime = new DateTime(period.getStartDate());
-      LocalDate startDate =
-          Instant.ofEpochMilli(period.getStartDate().getTime())
-              .atZone(ZoneId.systemDefault())
-              .toLocalDate();
-      LocalDate endDate =
-          Instant.ofEpochMilli(period.getEndDate().getTime())
-              .atZone(ZoneId.systemDefault())
-              .toLocalDate();
-      WeekFields weekFields = WeekFields.of(PeriodType.MAP_WEEK_TYPE.get(periodType.getName()), 4);
-      String year = String.valueOf(startDate.get(weekFields.weekBasedYear()));
-      String week = String.valueOf(startDate.get(weekFields.weekOfWeekBasedYear()));
-
-      if (isWeeklyPeriodType(periodType)) {
-        return String.format(
-            shortVersion
-                ? "W%s %d-%02d-%02d - %d-%02d-%02d"
-                : "Week %s %d-%02d-%02d - %d-%02d-%02d",
-            week,
-            startDate.getYear(),
-            startDate.getMonth().getValue(),
-            startDate.getDayOfMonth(),
-            endDate.getYear(),
-            endDate.getMonth().getValue(),
-            endDate.getDayOfMonth());
+      } else {
+        start = calendar.fromIso(period.getStartDate());
+        end = calendar.fromIso(period.getEndDate());
+        week = String.valueOf(calendar.week(start));
```

Comment:
Unclear: I would read the change to the initialization of the 3 fields as it being important that the `calendar` is used for some reason. What is that reason? Why does it not apply to the other branch of the if?

Reply:
the `calendar` is used when dealing with non `iso` calendar. We wanted to know which `calendar` so that we generate the date based on this calendar. The other branch of the `if` uses just period start and end dates which are based on `iso` calendar.

Reply:
Same here. I wanted the code to make it clear why it uses different approaches, something like "Note, this needs to use ISO because..."

Reply:
sure, have put a comment documenting the reason behond.